src/dscope.d

```diff
@@ -544,12 +544,27 @@ struct Scope
 
         s = searchScopes(flags | SearchImportsOnly);     // look in imported modules
 
-        version (LOGSEARCH)
+        if (s)
         {
-            if (s)
+            version (LOGSEARCH)
                 printMsg("-Scope.search() found import", s);
-            else
-                printf("-Scope.search() not found\n");
+            return s;
+        }
+
+        /** Still find private symbols, so that symbols that weren't access
+         * checked by the compiler remain usable.  Once the deprecation is over,
+         * this should be moved to search_correct instead.
+         */
+        s = searchScopes(flags | SearchLocalsOnly | IgnoreSymbolVisibility);
```

Comment:
> Put check for IgnorErrors and useDeprecated==1 here to avoid unnecessary extra symbol lookups.

I'd still have to return the correct symbol, even if not reporting the deprecation warning.
I can try to gather some stats how often symbol lookup falls through to this old behavior.